src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp

```diff
@@ -524,6 +524,20 @@ struct boss_cthun : public BossAI
             return;
         }
 
+        if (IsEveryoneAffectedByDigestiveAcid())
+        {
+            _acidTimer += diff;
```

Comment:
Instead of implementing your own timer, have you considered using the built-in scheduler? If `IsEveryoneAffectedByDigestiveAcid()` is true, you can schedule killing everyone on a 10-second delay. Then if `IsEveryoneAffectedByDigestiveAcid()` becomes false, cancel the scheduled execution. Seems like that would be better than re-inventing the wheel here.